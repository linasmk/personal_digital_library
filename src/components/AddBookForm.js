/* ============ Dependencies ============= */
import React from "react";
import Modal from "react-modal";
import { AiOutlineClose} from "react-icons/ai";
/* ============== Redux =============== */

/* ============ Components ============== */
/* =============== Code ================= */
//Modal.setAppElement("#app");
class AddBookForm extends React.Component {
  state = {
    title: this.props.book ? this.props.book.title : "",
    author: this.props.book ? this.props.book.author : "",
    description: this.props.book ? this.props.book.description : "",
    publicationYear: this.props.book ? this.props.book.publicationYear : "",
    publisher: this.props.book ? this.props.book.publisher : "",
    publicationLength: this.props.book ? this.props.book.publicationLength : "",
    category: this.props.book ? this.props.book.category : "",
    url: this.props.book ? this.props.book.url : "",
    rating: this.props.book ? this.props.book.rating : "",
    imgLink: this.props.book ? this.props.book.imgLink : "",
    error: "",
  };
  onTitleChange = (e) => {
    const title = e.target.value;
    this.setState(() => ({
      title,
    }));
  };
  onAuthorChange = (e) => {
    const author = e.target.value;
    this.setState(() => ({
      author,
    }));
  };
  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({
      description,
    }));
  };
  onPublicationYearChange = (e) => {
    const publicationYear = e.target.value;
    this.setState(() => ({
      publicationYear,
    }));
  };

  onPublicationLengthChange = (e) => {
    const publicationLength = e.target.value;
    this.setState(() => ({
      publicationLength,
    }));
  };

  onPublisherChange = (e) => {
    const publisher = e.target.value;
    this.setState(() => ({
      publisher,
    }));
  };

  onUrlChange = (e) => {
    const url = e.target.value;
    this.setState(() => ({
      url,
    }));
  };
  
  onCategoryChange = (e) => {
    const category = e.target.value;
    this.setState(() => ({
      category,
    }));
  };
  
  onRatingChange = (e) => {
    const rating = e.target.value;
    this.setState(() => ({
      rating,
    }));
  };
  onImgLinkChange = (e) => {
    const imgLink = e.target.value;
    this.setState(() => ({
      imgLink,
    }));
  };
  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.title) {
      this.setState(() => ({
        error: "Title's name field should not stay empty!",
      }));
    } else {
      this.setState(() => ({
        error: "",
      }));
      this.props.onSubmit({
        title: this.state.title,
        author: this.state.author,
        description: this.state.description,
        publicationYear: this.state.publicationYear,
        publisher: this.state.publisher,
        publicationLength: this.state.publicationLength,
        category: this.state.category,
        url: this.state.url,
        rating: this.state.rating,
        imgLink: this.state.imgLink,
      });
      this.props.clearAddNewBookModal();
    }
  };
  render() {
    return (
      <Modal
        isOpen={this.props.addNewBookModal}
        onRequestClose={this.props.clearAddNewBookModal}
        contentLabel={"Add New Book"}
        closeTimeoutMS={800}
        className="modal"
        ariaHideApp={false}
      >
      <div className="top-bar h-top-bar-modal">
        <AiOutlineClose
            className="card__btn-close h-icon-styles"
            viewBox="100 50 800 880"
            onClick={this.props.clearAddNewBookModal}
          />
      </div>
        <h2 className="addbook__heading">Add New Book</h2>
        <form className="addbook__form" onSubmit={this.onSubmit}>
          {this.state.error && (
            <p className="addbook-form__warning">{this.state.error}</p>
          )}
          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="text"
                required
                autoComplete="off"
                value={this.state.title}
                onChange={this.onTitleChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Book Title</span>
              </label>
            </div>
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="text"
                required
                autoComplete="off"
                value={this.state.author}
                onChange={this.onAuthorChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Author</span>
              </label>
            </div>
          </div>
          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="number"
                required
                value={this.state.publicationYear}
                onChange={this.onPublicationYearChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Publication Year</span>
              </label>
            </div>
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="number"
                required
                value={this.state.publicationLength}
                onChange={this.onPublicationLengthChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">
                  Publication Length
                </span>
              </label>
            </div>
          </div>

          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="text"
                required
                value={this.state.publisher}
                onChange={this.onPublisherChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Publisher</span>
              </label>
            </div>
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="url"
                title="URL should match the pattern: https://.*"
                required
                pattern="https://.*"
                autoComplete="off"
                value={this.state.url}
                onChange={this.onUrlChange}
              />
              <label htmlFor="url" className="addbook__form_label">
                <span className="addbook__form_content">Link to the Book</span>
              </label>
            </div>
          </div>

          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="text"
                required
                value={this.state.category}
                onChange={this.onCategoryChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Category</span>
              </label>
            </div>

            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="number"
                min="1"
                max="10"
                required
                value={this.state.rating}
                onChange={this.onRatingChange}
              />
              <label htmlFor="title" className="addbook__form_label">
                <span className="addbook__form_content">Rating</span>
              </label>
            </div>
          </div>
          <div className="addbook__form_row">
            <div className="addbook__form_column">
              <input
                className="addbook__form_input"
                type="url"
                title="URL should match the pattern: https://.*"
                required
                autoComplete="on"
                value={this.state.imgLink}
                onChange={this.onImgLinkChange}
              />
              <label htmlFor="url" className="addbook__form_label">
                <span className="addbook__form_content">Image Link</span>
              </label>
            </div>
          </div>
          <div className="addbook__form_row">
            <div className="addbook__form_column-textarea">
              <textarea
                className="addbook__form_textarea"
                type="text"
                required
                maxLength="3000"
                value={this.state.description}
                onChange={this.onDescriptionChange}
              />
              <label htmlFor="title" className="addbook__form_label-textarea">
                <span className="addbook__form_content-textarea">
                  Book Description
                </span>
              </label>
            </div>
          </div>
          <div className="modal__button-bottom_wrapper">
            <button className="modal__button-bottom" type="submit">
              Add Book
            </button>
          </div>
        </form>
      </Modal>
    );
  }
}

export default AddBookForm;
