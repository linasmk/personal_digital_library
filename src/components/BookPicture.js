import React from "react";
const BookPicture = (props) => {
  return (
    <picture>
      <source media="(min-width: 1400px)" srcSet={props.srcXlg} sizes="250px" />
      <source media="(min-width: 998px)" srcSet={props.srcLg} sizes="200px" />
      <source media="(min-width: 480px)" srcSet={props.srcMd} sizes="120px" />
      <source media="(min-width: 200px)" srcSet={props.srcSm} sizes="80px" />
      <img className="book__picture" src={props.src} alt="title" />
    </picture>
  );
};
export default BookPicture;
