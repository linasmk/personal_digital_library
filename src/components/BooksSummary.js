/* ========= App Dependencies ============= */
import React from "react";
import { connect } from "react-redux";
/* ========== Redux ========== */
import getVisibleBooks from "../store/selectors";

/* ========= Code ============= */
export const BooksSummary = ({ booksCount }) => {
  return (
    <section className="header__summary_block">
      <p className="header__summary_p">
        Filtered books:{" "}
        <span className="header__summary_output">{booksCount}</span>
      </p>
    </section>
  );
};

const mapStateToProps = (state) => {
  const visibleBooks = getVisibleBooks(state.books, state.filters);
  return {
    booksCount: visibleBooks.length,
  };
};

export default connect(mapStateToProps)(BooksSummary);
