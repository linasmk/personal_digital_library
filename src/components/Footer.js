import React from "react";
const Footer = props => {
  const currentYear = new Date().getFullYear();
  
      
  
  return (
    <footer className="footer">
      <div className="footer__overlay">
        <h3 className="footer__copyright">
          &copy; {currentYear}. Linas Mackonis
        </h3>
      </div>
    </footer>
  );
};
export default Footer;
