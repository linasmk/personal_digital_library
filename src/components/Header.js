import React from "react";
const Header = props => {
	return (<header>
      <h1 className="main-heading">{props.title}</h1>
      <h2>{props.handleCopyrightDate}</h2>
    </header>);
};
Header.defaultProps = {
	title: "Personal Digital Library"
};
export default Header;