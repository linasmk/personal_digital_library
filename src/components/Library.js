import React from "react";
const library = [
  {
    id: 2000,
    title:
      "Gamify: How Gamification Motivates People to Do Extraordinary Things",
    author: "Brian Burke",
    description:
      "Gamify shows gamification in action: as a powerful approach to engaging and motivating people to achieving their goals, while at the same time achieving organizational objectives. It can be used to motivate people to change behaviors, develop skills, and drive innovation. The key to gamification success is to engage people on an emotional level and motivating them to achieve their goals.",
    publicationYear: 2014,
    publicationLength: 195,
    category: "business",
    url:
      "https://www.amazon.com/Gamify-Gamification-Motivates-People-Extraordinary-ebook/dp/B00I2ZW47W",
    rating: 8,
    src: "img/book_covers/gamify324x500.jpg",
    srcSm: "img/book_covers/gamify80x123.jpg",
    srcMd: "img/book_covers/gamify120x185.jpg",
    srcLg: "img/book_covers/gamify200x307.jpg",
    srcXlg: "img/book_covers/gamify250x386.jpg",
    background:
      "linear-gradient(185deg, rgb(0, 255, 255), rgb(255, 255, 255), rgb(0, 255, 255));"
  },
  {
    id: 5000,
    title: "Mind As Machine: A History of Cognitive Science Two-Volume Set",
    author: "Margaret Boden",
    description:
      "Psychology is the thematic heart of cognitive science, which aims to understand human (and animal) minds. But its core theoretical ideas are drawn from cybernetics and artificial intelligence, and many cognitive scientists try to build functioning models of how the mind works. In that sense, Margaret Boden suggests, its key insight is that mind is a (very special) machine. Because the mind has many different aspects, the field is highly interdisciplinary. It integrates psychology not only with cybernetics/AI, but also with neuroscience and clinical neurology; with the philosophy of mind, language, and logic.",
    publicationYear: 2000,
    publicationLength: 1712,
    category: "cognitive science",
    url: "https://www.amazon.com/Mind-As-Machine-Cognitive-Two/dp/0199241449",
    rating: 8,
    src: "img/book_covers/mind_as_machine396x499.jpg",
    srcSm: "img/book_covers/mind_as_machine80x101.jpg",
    srcMd: "img/book_covers/mind_as_machine120x151.jpg",
    srcLg: "img/book_covers/mind_as_machine200x252.jpg",
    srcXlg: "img/book_covers/mind_as_machine250x315.jpg",
    background:
      "linear-gradient(185deg, rgb(0, 255, 255), rgb(255, 255, 255), rgb(0, 255, 255));"
  },
  {
    id: 5001,
    title: "Bottlenecks: Aligning UX Design with User Psychology",
    author: "David C. Evans",
    description:
      "This book fills a need for entrepreneurs, designers, and marketing professionals in the application of foundational psychology to user-experience design. The first generation of books on the topic focused on web pages and cognitive psychology. This book covers apps, social media, in-car infotainment, and multiplayer video games, and it explores the crucial roles played by behaviorism, development, personality, and social psychology.",
    publicationYear: 2017,
    publicationLength: 284,
    category: "cognitive science",
    url:
      "https://www.amazon.com/Bottlenecks-Aligning-Design-User-Psychology-ebook-dp-B06W2J32CK/dp/B06W2J32CK/ref=mt_kindle?_encoding=UTF8&me=&qid=",
    rating: 9,
    src: "img/book_covers/bottlenecks330x499.jpg",
    srcSm: "img/book_covers/bottlenecks80x120.jpg",
    srcMd: "img/book_covers/bottlenecks120x181.jpg",
    srcLg: "img/book_covers/bottlenecks200x302.jpg",
    srcXlg: "img/book_covers/bottlenecks250x379.jpg",
    background:
      "linear-gradient(185deg, rgb(0, 255, 255), rgb(255, 255, 255), rgb(0, 255, 255));"
  },
  {
    id: 6000,
    title: "JavaScript: The Good Parts",
    author: "Douglas Crockford",
    description:
      "Most programming languages contain good and bad parts, but JavaScript has more than its share of the bad, having been developed and released in a hurry before it could be refined. This authoritative book scrapes away these bad features to reveal a subset of JavaScript that's more reliable, readable, and maintainable than the language as a whole subset you can use to create truly extensible and efficient code.",
    publicationYear: 2008,
    publicationLength: 176,
    category: "programming languages",
    url:
      "https://www.amazon.com/JavaScript-Good-Parts-Douglas-Crockford/dp/0596517742",
    rating: 8,
    src: "img/book_covers/js_the_good_parts383x499.jpg",
    srcSm: "img/book_covers/js_the_good_parts80x104.jpg",
    srcMd: "img/book_covers/js_the_good_parts120x156.jpg",
    srcLg: "img/book_covers/js_the_good_parts200x260.jpg",
    srcXlg: "img/book_covers/js_the_good_parts250x325.jpg",
    background:
      "linear-gradient(185deg, rgb(0, 255, 255), rgb(255, 255, 255), rgb(0, 255, 255));"
  },
  {
    id: 1000,
    title: "The Power of Habit: Why We Do What We Do in Life and Business",
    author: "Charles Duhigg",
    description:
      "Duhigg presents a whole new understanding of human nature and its potential. At its core, The Power of Habit contains an exhilarating argument: The key to exercising regularly, losing weight, being more productive, and achieving success is understanding how habits work. As Duhigg shows, by harnessing this new science, we can transform our businesses, our communities, and our lives.",
    publicationYear: 2012,
    publicationLength: 400,
    category: "psychology",
    url:
      "https://www.amazon.com/Power-Habit-What-Life-Business/dp/1400069289/ref=tmm_hrd_swatch_0?_encoding=UTF8&qid=1582294939&sr=1-1",
    rating: 7.9,
    src: "img/book_covers/the_power_of_habit329x499.jpg",
    srcSm: "img/book_covers/the_power_of_habit80x121.jpg",
    srcMd: "img/book_covers/the_power_of_habit120x182.jpg",
    srcLg: "img/book_covers/the_power_of_habit200x303.jpg",
    srcXlg: "img/book_covers/the_power_of_habit250x379.jpg",
    background:
      "linear-gradient(185deg, rgb(0, 255, 255), rgb(255, 255, 255), rgb(0, 255, 255));"
  }
];
export default library;
