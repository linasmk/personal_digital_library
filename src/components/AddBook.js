import React, { Component } from "react";
import { FaPlus } from "react-icons/fa";
import AddBookForm from "./AddBookForm";

export default class AddBook extends Component {
  state = {
    formDisplay: undefined,
  };
  handleAddBookFormPopUp = () => {
    this.setState(() => ({
      formDisplay: true,
    }));
  };
  render() {
    return (
      <section className="addbook__wraper">
        <button
          className="button__addbook"
          onClick={this.handleAddBookFormPopUp.bind(this)}
        >
          <FaPlus className="button__addbook_icon" />
        </button>
        <AddBookForm formDisplay={this.state.formDisplay} />
      </section>
    );
  }
}
