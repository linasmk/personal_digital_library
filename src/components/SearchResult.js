/* =============== Dependencies ============= */
import React, { useState } from "react";
import { FaRegTrashAlt } from "react-icons/fa";

/* ========== Redux ============= */
import { connect } from "react-redux";
import getVisibleBooks from "../store/selectors";
import { removeBook, editBook } from "../store/actionCreators";

/* ============== Components ================= */
import BookPicture from "./BookPicture";
import ModalBox from "./ModalBox";
import StarRating from "./StarRating";

/* =============== Code ======================== */
const SearchResult = (props) => {
  const [viewBookModalId, setBookModallId] = useState(undefined);
  const viewBookModalPopUp = (id) => setBookModallId(id);
  const clearViewBookModal = () => setBookModallId(undefined);

  const [starRating, setStarRating] = useState(0);

  const handleStarRating = (rating, starRating, id) => {
    if (starRating === rating) {
      setStarRating(0);
    } else {
      setStarRating({ starRating });
      setStarRating(() => props.dispatch(editBook(id, { starRating })));
    }
  };

  // const handleStarRating = (rating) => {
  //   if (this.state.rating === rating) {
  //     this.setState({ rating: 0 });
  //   } else {
  //     this.setState({ rating });
  //     this.setState(() =>
  //       this.props.dispatch(editBook(this.state.id, { rating }))
  //     );
  //   }
  // };

  return (
    <div className="search__result">
      {props.books.map((book) => (
        <div key={book.id} className="card__container">
          <article className="card">
            <div className="top-bar h-top-bar-search-results">
              <FaRegTrashAlt
                className="card__btn-trash h-icon-styles"
                onClick={(e) => props.dispatch(removeBook({ id: book.id }))}
              />
            </div>

            <section className="card__main-block">
              <a className="card__book_link" href={book.url} target="_blank">
                <BookPicture
                  src={book.src}
                  srcSm={book.srcSm}
                  srcMd={book.srcMd}
                  srcLg={book.srcLg}
                  srcXlg={book.srcXlg}
                />
              </a>

              <h2 className="card__book_title">{book.title}</h2>
              <h3 className="card__book_author">
                <span>by </span>
                {book.author}
              </h3>
              <StarRating
                // handleStarRating={handleStarRating}
                bookRating={book.rating}
                bookId={book.id}
                // Star props
                starRatingUl="star-rating__ul"
                star="star"
                selected="selected"
                stroke={undefined}
                strokeWidth={0}
              />
              <div className="view__more_box">
                <button
                  className="view__more_btn"
                  onClick={() => viewBookModalPopUp(book.id)}
                >
                  View More
                </button>
              </div>
            </section>
          </article>
          <ModalBox
            //Book info
            key={book.id}
            handleStarRatin={handleStarRating}
            {...book}
            // Modal handlers
            clearViewBookModal={clearViewBookModal}
            viewBookModal={viewBookModalId === book.id}
          />
        </div>
      ))}
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    books: getVisibleBooks(state.books, state.filters),
  };
};
export default connect(mapStateToProps)(SearchResult);
