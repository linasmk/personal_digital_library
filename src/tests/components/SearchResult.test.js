import React from "react";
import { shallow } from "enzyme";
import { SearchResult } from "../../components/SearchResult";
import books from "../fixtures/books";
import { wrap } from "lodash";

test("Should render books list with books", () => {
  const wrapper = shallow(<SearchResult books={books} />);
  expect(wrapper).toMatchSnapshot();
});

test("Should render SearchResult with empty message", () => {
  const wrapper = shallow(<SearchResult books={[]} />);
  expect(wrapper).toMatchSnapshot();
});
