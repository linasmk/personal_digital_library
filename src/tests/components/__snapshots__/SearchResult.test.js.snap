// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render SearchResult with empty message 1`] = `
<div
  className="search__result"
>
  <h3>
    You currently do not have any books!
  </h3>
</div>
`;

exports[`Should render books list with books 1`] = `
<div
  className="search__result"
>
  <div
    className="card__container"
    key="1"
  >
    <article
      className="card"
    >
      <div
        className="top-bar h-top-bar-search-results"
      >
        <FaRegTrashAlt
          className="card__btn-trash h-icon-styles"
          onClick={[Function]}
        />
      </div>
      <section
        className="card__main-block"
      >
        <a
          className="card__book_link"
          target="_blank"
        >
          <BookPicture />
        </a>
        <h2
          className="card__book_title"
        >
          Why Zebras Dont't Get Ulcers
        </h2>
        <h3
          className="card__book_author"
        >
          <span>
            by 
          </span>
          Robert Sapolski
        </h3>
        <Connect(StarRating)
          bookId="1"
          selected="selected"
          star="star"
          starRatingUl="star-rating__ul"
          strokeWidth={0}
        />
        <div
          className="view__more_box"
        >
          <button
            className="view__more_btn"
            onClick={[Function]}
          >
            View More
          </button>
        </div>
      </section>
    </article>
    <Connect(ModalBox)
      author="Robert Sapolski"
      clearViewBookModal={[Function]}
      handleStarRatin={[Function]}
      id="1"
      key="1"
      title="Why Zebras Dont't Get Ulcers"
      viewBookModal={false}
    />
  </div>
  <div
    className="card__container"
    key="2"
  >
    <article
      className="card"
    >
      <div
        className="top-bar h-top-bar-search-results"
      >
        <FaRegTrashAlt
          className="card__btn-trash h-icon-styles"
          onClick={[Function]}
        />
      </div>
      <section
        className="card__main-block"
      >
        <a
          className="card__book_link"
          target="_blank"
        >
          <BookPicture />
        </a>
        <h2
          className="card__book_title"
        >
          Incognito: The Secret Lives of the Brain
        </h2>
        <h3
          className="card__book_author"
        >
          <span>
            by 
          </span>
          David Eagleman
        </h3>
        <Connect(StarRating)
          bookId="2"
          selected="selected"
          star="star"
          starRatingUl="star-rating__ul"
          strokeWidth={0}
        />
        <div
          className="view__more_box"
        >
          <button
            className="view__more_btn"
            onClick={[Function]}
          >
            View More
          </button>
        </div>
      </section>
    </article>
    <Connect(ModalBox)
      author="David Eagleman"
      clearViewBookModal={[Function]}
      handleStarRatin={[Function]}
      id="2"
      key="2"
      title="Incognito: The Secret Lives of the Brain"
      viewBookModal={false}
    />
  </div>
  <div
    className="card__container"
    key="3"
  >
    <article
      className="card"
    >
      <div
        className="top-bar h-top-bar-search-results"
      >
        <FaRegTrashAlt
          className="card__btn-trash h-icon-styles"
          onClick={[Function]}
        />
      </div>
      <section
        className="card__main-block"
      >
        <a
          className="card__book_link"
          target="_blank"
        >
          <BookPicture />
        </a>
        <h2
          className="card__book_title"
        >
          Thinking, Fast and Slow
        </h2>
        <h3
          className="card__book_author"
        >
          <span>
            by 
          </span>
          Daniel Kahneman
        </h3>
        <Connect(StarRating)
          bookId="3"
          selected="selected"
          star="star"
          starRatingUl="star-rating__ul"
          strokeWidth={0}
        />
        <div
          className="view__more_box"
        >
          <button
            className="view__more_btn"
            onClick={[Function]}
          >
            View More
          </button>
        </div>
      </section>
    </article>
    <Connect(ModalBox)
      author="Daniel Kahneman"
      clearViewBookModal={[Function]}
      handleStarRatin={[Function]}
      id="3"
      key="3"
      title="Thinking, Fast and Slow"
      viewBookModal={false}
    />
  </div>
</div>
`;
